specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/apimportalrevisions.json

```diff
@@ -0,0 +1,326 @@
+            }
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "PortalRevision"
+        ],
+        "operationId": "PortalRevision_Get",
+        "description": "Gets developer portal revision specified by its identifier.",
+        "x-ms-examples": {
+          "ApiManagementGetPortalRevision": {
+            "$ref": "./examples/ApiManagementGetPortalRevision.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/PortalRevisionIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Gets the specified developer portal revision entity.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/PortalRevisionContract"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "PortalRevision"
+        ],
+        "operationId": "PortalRevision_CreateOrUpdate",
+        "description": "Creates a new developer portal revision.",
+        "x-ms-examples": {
+          "ApiManagementCreatePortalRevision": {
+            "$ref": "./examples/ApiManagementCreatePortalRevision.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/PortalRevisionIdParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "./definitions.json#/definitions/PortalRevisionContract"
+            }
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "202": {
```

Comment:
async put utilizes 201/200 + provisioningState.  See https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creatingupdating-using-put

Reply:
Response 202 return then we accepted the request start to create a revision. 
This is a long-running operation but we do not provision anything, mostly generating files for the current content state. 
Usually, it will take time from 5 sec up to several minutes. To track the portal revision state we use property Status. It can be  "pending", "publishing", "completed", "failed". We have the same for NamedValues resource and do not have  **provisioningState**.

Reply:
This appears to be a less commonly used "final-state-via": "location" scheme. SDK will poll for 202, until it is not 202 (usually 200/201/204 or error).

Reply:
Yeah according to this part of the RPC, it seems to be allowed:
![image](https://user-images.githubusercontent.com/22460039/100182098-a8608c00-2e90-11eb-9cd3-d8f7bc7e5911.png)